matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/database/model/EventEntity.kt

```diff
@@ -21,11 +21,10 @@ import im.vector.matrix.android.internal.crypto.MXEventDecryptionResult
 import im.vector.matrix.android.internal.crypto.algorithms.olm.OlmDecryptionResult
 import im.vector.matrix.android.internal.di.MoshiProvider
 import io.realm.RealmObject
-import io.realm.RealmResults
 import io.realm.annotations.Index
-import io.realm.annotations.LinkingObjects
+import io.realm.annotations.PrimaryKey
 
-internal open class EventEntity(@Index var eventId: String = "",
+internal open class EventEntity(@PrimaryKey var eventId: String = "",
```

Comment:
Why moving to Primary key? what's the advantage?

Reply:
I prefer to ensure event entity are really constrained. It's to avoid adding multiple eventEntity when adding TimelineEvent and other of entities linked to EventEntity.